### PR TITLE
Axios added to dependencies and implemented in FlightTracker class

### PR DESCRIPTION
**Pull request by DeathAdderR**
- Added axios to the project dependencies to make HTTP requests
- Implemented logic within the bin.js file for CLI commands to handle airport and flight search functionality
- Instantiated axios within the FlightTrackerCLI class to manage API requests to Aviationstack.

### CODE REVIEW

File: bin/flightTracker.js [hunk 1]
```diff
@@ -46,10 +49,21 @@ var options = yargs(process.argv.slice(2))
...
+      })
+
+      // Will we be passing an access_key to the API?
+
+      // .option("access_key", {
+      //   alias: "k",
+      //   description: "You AviationStack API access key",
+      //   type: "string",
+      //   demandOption: true
+      // })
```
Comment (mdwiltfong) [inline bin/flightTracker.js:61]: This is a good question. The user probably can't run this command until they have fetched the API key, or have manually passed it to CLI. In hindsight I should have changed the `login()` function to `fetchAPIKey()` for when they are using BitWarden.

File: bin/flightTracker.js [hunk 2]
```diff
@@ -46,10 +49,21 @@ var options = yargs(process.argv.slice(2))
...
     },
     (argv) => {
-      if (argv["name"]) {
+      if (argv['name']) {
+        // REQUIRES API KEY
+        // flightTracker.findIATACode(argv['name'])
```
Comment (mdwiltfong) [inline bin/flightTracker.js:66]: This functiona will require the API key. If the user hasn't passed the key, or they haven't fetched it from BitWarden, then this function needs to throw an error and prompt the user to pick one of the two approaches to getting the key.

File: bin/utils/utils.js [hunk 3]
```diff
@@ -1,7 +1,74 @@
...
+      }
+    });
+  }
+
+  async findIATACode(airportName) {
+    try {
+      const response = await this.axiosInstance.get('/airports', {
+        params: {
+          city: airportName,
+          // access_key: 'YOUR_API_KEY', // option to pass key manually
```
Comment (mdwiltfong) [inline bin/utils/utils.js:21]: We can add the api key as a class property. If this property is null, we should throw an error telling the user no API key has been found and that they need to run the appropriate commands.

File: bin/utils/utils.js [hunk 4]
```diff
@@ -1,7 +1,74 @@
...
+        'Content-Type': 'application/json',
+      }
+    });
+  }
+
+  async findIATACode(airportName) {
+    try {
+      const response = await this.axiosInstance.get('/airports', {
+        params: {
+          city: airportName,
```
Comment (mdwiltfong) [inline bin/utils/utils.js:20]: We should probably convert this user input into lowercase and trim any whitespace.

File: bin/utils/utils.js [hunk 5]
```diff
@@ -1,7 +1,74 @@
...
+      }
+    } catch (error) {
+      console.error('Error fetching flights:', error.message || error)
+    }
+  }
 
-  login(manualLogin = false, userName = "", passWord = "") {}
+  async login(manualLogin = false, userName = "", passWord = "") {
+    if (manualLogin) {
+      // manual logic
```
Comment (mdwiltfong) [inline bin/utils/utils.js:66]: Thanks for writing these notes. I'm still toying around on how to handle manually passing the api key. Any ideas would be appreciated here #8

**Review by mdwiltfong — CHANGES_REQUESTED** [review]
Thanks for this awesome PR @DeathAdderR ! I started this as an idea for the weekly JS Support Hours, and I was also excited to use it to engage with learners. So excuse me if somethings aren't clear. You wrote good code despite my lack of clarity. 

I add some comments where I can add more information. Let me know if you questions. This contribution is also good as it is. So if you don't feel tempted to follow-up on these comments, please feel free to let me know and I can roll with what you wrote so far. When the PR gets merged you'll be labeled as a co-author and get credit for contributing either way.

File: bin/utils/utils.js (hunk 3 above)
Comment (DeathAdderR) [reply bin/utils/utils.js:21]: Ahh yeah that is really good.

File: bin/utils/utils.js (hunk 4 above)
Comment (DeathAdderR) [reply bin/utils/utils.js:20]: we now pass an instance of axios to the constructor and a default null value for apiKey

File: bin/flightTracker.js (hunk 2 above)
Comment (DeathAdderR) [reply bin/flightTracker.js:69]: Now the flightTracker's methods check for an apiKey.

**Review by mdwiltfong — CHANGES_REQUESTED** [review]

**Pull request closed without merging**